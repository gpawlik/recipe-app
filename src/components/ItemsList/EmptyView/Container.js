import styled from 'styled-components';

const Container = styled.ul`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export default Container;
