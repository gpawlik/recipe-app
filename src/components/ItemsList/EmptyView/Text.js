import styled from 'styled-components';

import { colors } from 'variables';

const Container = styled.p`
  color: ${colors.CONCRETE_GRAY};
  font-size: 20px;
  font-family: 'Helvetica Neue', Arial;
`;

export default Container;
