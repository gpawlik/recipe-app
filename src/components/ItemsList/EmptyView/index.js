import React from 'react';
import PropTypes from 'prop-types';

import Container from './Container';
import Text from './Text';

const EmptyView = ({ text }) => {
  return (
    <Container>
      <Text>{text}</Text>
    </Container>
  );
};

EmptyView.defaultProps = {
  text: 'Sorry, no results found...'
};

EmptyView.propTypes = {
  text: PropTypes.string
};

export default EmptyView;
