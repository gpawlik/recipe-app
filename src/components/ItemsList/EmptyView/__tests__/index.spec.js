import React from 'react';
import { create } from 'react-test-renderer';

import EmptyView from '../';

const initialProps = {};

const mockComponent = props => {
  const finalProps = {
    ...initialProps,
    ...props
  };

  return (
    <EmptyView {...finalProps} />
  );
};

describe('EmptyView', () => {
  it('should render with the default props', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with custom text', () => {
    const props = {
      text: 'foo'
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
