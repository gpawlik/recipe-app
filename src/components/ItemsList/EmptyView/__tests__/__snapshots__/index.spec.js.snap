// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyView should render with custom text 1`] = `
<ul
  className="sc-bdVaJa grbsvW"
>
  <p
    className="sc-bwzfXH cstZNn"
  >
    foo
  </p>
</ul>
`;

exports[`EmptyView should render with the default props 1`] = `
<ul
  className="sc-bdVaJa grbsvW"
>
  <p
    className="sc-bwzfXH cstZNn"
  >
    Sorry, no results found...
  </p>
</ul>
`;
